src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/ViewContext.cs

```diff
@@ -58,24 +59,24 @@ public ViewContext()
                 throw new ArgumentNullException(nameof(view));
             }
 
-            if (viewData == null)
+            if (writer == null)
             {
-                throw new ArgumentNullException(nameof(viewData));
+                throw new ArgumentNullException(nameof(writer));
             }
 
-            if (tempData == null)
+            if (htmlHelperOptions == null)
             {
-                throw new ArgumentNullException(nameof(tempData));
+                throw new ArgumentNullException(nameof(htmlHelperOptions));
             }
 
-            if (writer == null)
+            if (viewData == null)
             {
-                throw new ArgumentNullException(nameof(writer));
+                throw new ArgumentNullException(nameof(viewData));
             }
 
-            if (htmlHelperOptions == null)
+            if (tempData == null)
             {
-                throw new ArgumentNullException(nameof(htmlHelperOptions));
+                throw new ArgumentNullException(nameof(tempData));
```

Comment:
Restore these `null` checks in their original order. That nicely matched the parameter order.